README.md

```diff
@@ -35,9 +35,25 @@ fetchBodyTypes()
 | `optionServiceUrl`           | URL to [Options Service](https://option-service.preprod.carforyou.ch/swagger-ui/index.html)                         |
 | `analyticsServiceUrl`        | URL to [Analytics Service](https://carforyou-analytics-service.preprod.carforyou.ch/swagger-ui/index.html)          |
 | `userNotificationServiceUrl` | URL to [User Notification Service](https://user-notification-service.preprod.carforyou.ch/swagger-ui/index.html)    |
-| `tokenRefreshServiceUrl`     | URL to Auth Service used to refresh access tokens                                                                   |
 | `debug`                      | Set to `true` to `console.log` requests and API responses.                                                          |
 
+### Authenticated Requests
+
+#### Access Token
+
+In order to perform authenticated requests you need to set a valid access token to the api client in the consumers code `ApiClient.setAccessToken(JWT)` The provided token will be passed as `Authentication` header to the API.
+
+#### Refreshing Access Tokens
+
+Authenticated requests will try to issue a refresh token when they revive a 401 response form the remote server. In order to handle the process of refreshing the token the consumer must pass a handler function to the API-Client. (The function may vary depending on the environment the API-Client is used)
+
+```
+ApiClient.setTokenRefreshHandler(async () => {
+  const response = await fetch(process.env.TOKEN_REFRESH_ROUTE)
+  return response.json()
+})
+```
```

Comment:
I might see the answer later in the code, but how is the case handled when this function is missing, but the API client runs into a 401. A helpful message would be appreciated over throwing a cryptic message :)

Reply:
will we have spot-on types for this functions return value to make sure we don't fuck refreshing up?

Reply:
1. My intention was to throw a meaningful error in the function added in the prototype https://github.com/carforyou/carforyou-api-client-pkg/pull/175/files#diff-c50cd8cbf813f34536ae74335bcf5724R56 so if one misses to add it the error shall point them to whats off

2. The function is typed pretty strict https://github.com/carforyou/carforyou-api-client-pkg/pull/175/files#diff-1c1a34a94ec474ad2092388c3650708eR65 let me know if you would add something on that.

Reply:
thanks for double checking. I forgot to delete this reminder-to-self-comment when I reached the function and answered the question to myself ;)